src/Behat/Behat/Output/Node/Printer/JUnit/JUnitSuitePrinter.php

```diff
@@ -44,7 +44,11 @@ public function printHeader(Formatter $formatter, Suite $suite)
 
         /** @var JUnitOutputPrinter $outputPrinter */
         $outputPrinter = $formatter->getOutputPrinter();
-        $outputPrinter->createNewFile($suite->getName());
+        if (!empty($formatter->fileName)) {
+          $outputPrinter->createNewFile($formatter->fileName);
```

Comment:
Could you use only one `createNewFile` call, such as:
```php
$outputPrinter->createNewFile($formatter->fileName ?: $suite->getName());
```
(or put the filename in a variable is you prefer.